### PR TITLE
Handle constants in more places

### PR DESCRIPTION
**Pull request by ltratt**
The first commit is what I *thought* needed fixing before I had a backtrace to know where @ptersilie was seeing a problem. Still, it seems useful, and it's related to the "real" fix in the second commit, so I think the first commit is worth keeping.

@ptersilie: This PR tickles some sensitive parts of the system and I haven't yet thought through every consequence. I *think* the changes are probably OK, but you might want to check this for your use case before considering merging.

### CODE REVIEW

File: tests/c/simple_interp_loop2.c [hunk 1]
```diff
@@ -12,7 +12,7 @@
 //     yk-jit-event: stop-tracing
 //     --- Begin jit-pre-opt ---
 //     ..~
-//     guard true, %{{34}}, [%{{0}}, %{{1}}, %{{2}}, %{{3}}, %{{4}}, %{{5}}, %{{34}}]
+//     guard true, %{{34}}, [2/0/2: %{{0}}, 2/0/4: %{{1}}, 2/0/5: %{{2}}, 2/0/6: %{{3}}, 2/0/7: %{{4}}, 2/0/9: %{{5}}, {{_}}/23/3: %{{34}}]
```
[the remark this answers is not in the document]
Comment (ltratt) [reply tests/c/simple_interp_loop2.c]: I don't know about you, but I *continually* forget what the arguments to a guard are. Indeed, the fact that this was only fixed a few days ago https://github.com/ykjit/yk/pull/1377/commits/90ff056cad2fa70599d1d9affffbe8dcfd82338c suggests I'm not the only one who misleads myself here!

I agree with the formatting consistency change. Fixed in https://github.com/ykjit/yk/pull/1377/commits/90ff056cad2fa70599d1d9affffbe8dcfd82338c, 8cda14cbff52efd09875b16e3b89a05f8c2c212f, and 0cd2f67e8c09ba7d4950755309e4ec5f524df96f.

[the remark this answers is not in the document]
Comment (ptersilie) [reply tests/c/simple_interp_loop2.c]: Okay, let's roll with this for now and see how it goes.

**Comment by ltratt** [comment]
Test (hopefully) fixed in 792a36853d342343e85a83185b6e32ebe847ba04.

File: tests/c/simple_interp_loop2.c [hunk 2]
```diff
@@ -12,7 +12,7 @@
 //     yk-jit-event: stop-tracing
 //     --- Begin jit-pre-opt ---
 //     ..~
-//     guard true, %{{34}}, [2:%0_2: %{{0}}, 2:%0_4: %{{1}}, 2:%0_5: %{{2}}, 2:%0_6: %{{3}}, 2:%0_7: %{{4}}, 2:%0_9: %{{5}}, {{_}}:%23_3: %{{34}}]
+//     guard true, %34, [2:%0_{{_}}: %0, 2:%0_{{_}}: %1, 2:%0_{{_}}: %2, 2:%0_{{_}}: %3, 2:%0_{{_}}: %4, 2:%0_{{_}}: %5, 2:%23_{{_}}: %34]
```
Comment (ptersilie) [inline tests/c/simple_interp_loop2.c:15]: I think `%34` should stay `%{{34}}` otherwise the test fails if the IR numbers change even slightly.

Comment (ltratt) [reply tests/c/simple_interp_loop2.c:15]: OK. Shall I squash that change straight in?

Comment (ltratt) [reply tests/c/simple_interp_loop2.c]: Erp. Found another way the test can fail. Fixed in a0ba4fefebe62fe08c45126401b25dc997c17ce3.

**Comment by ptersilie** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**